### PR TITLE
Enhanced route refresh capability improvements

### PR DESCRIPTION
**Pull request by ton31337**

### CODE REVIEW

**Comment by ton31337** [comment]
@infrastation do you mind reviewing this, please?

**Comment by infrastation** [comment]
Please have patience, there is a lot of backlog. Maybe another maintainer will have time to proof-read the changes.

**Review by fenner — COMMENTED** [review]
I think you need to improve the title / description.  (The title is probably left over from just changing `bgp_capabilities_print()`).

How about adding parsing for the new NOTIFICATION too, and its subcode?
```
5.  Error Handling

   This document defines a new NOTIFICATION error code:

          Error Code   Name

              7        ROUTE-REFRESH Message Error

   The following error subcode is defined as well:

          Subcode      Name

             1         Invalid Message Length
```


It might need to be rebased, I'll leave that to the actual committer.  I'm just reviewing the code.

File: tests/bgp-enhanced-route-refresh-subtype.out
Comment (ton31337) [reply tests/bgp-enhanced-route-refresh-subtype.out:4]: Will add this.

Comment (ton31337) [reply tests/bgp-enhanced-route-refresh-subtype.out:3]: Added.

**Comment by ton31337** [comment]
@fenner all good now?

**Comment by fenner** [comment]
> @fenner all good now?

Did you see my suggestion to implement the new NOTIFICATION and its subtype?

**Comment by ton31337** [comment]
> > @fenner all good now?
> 
> Did you see my suggestion to implement the new NOTIFICATION and its subtype?

I saw. Let's just get this separate. I'll add another PR for error codes to distinguish things a bit and keep them as small as possible.

**Review by fenner — APPROVED** [review]
This looks like a good parser for most of RFC7313.  I hope to see a NOTIFICATION parser in the future, but this code looks good to me to merge as-is.

**Comment by ton31337** [comment]
@fxlb @infrastation just rebased to the latest master (was a conflict with TESTLIST file). Can be merged?

**Comment by fxlb** [comment]
Could you first 'rebase -i' it to have a clean commit?

**Comment by ton31337** [comment]
Do you mean squash into a single commit or what specifically?

**Comment by fxlb** [comment]
Any reason to have more than one commit?

**Comment by ton31337** [comment]
Don't know the flow and backporting logic. If that's not necessary, I can easily squash, just tell me to squash or not :)

**Comment by ton31337** [comment]
Anyway, squashed :heavy_check_mark:

**Comment by fxlb** [comment]
Rethinking about this PR, it's seems better with 3 commits (codes with corresponding tests). Done.

**Comment by fxlb** [comment]
Last question before merge: It's seems better to update the message for the second commit:
`tests: No need to decode enhanced route refresh capability`
to
`bgp: <some text like Add a test file for...>`
With a `bgp:` prefix. What message would be relevant?
(I will do the update)

**Comment by ton31337** [comment]
> Last question before merge: It's seems better to update the message for the second commit: `tests: No need to decode enhanced route refresh capability` to `bgp: <some text like Add a test file for...>` With a `bgp:` prefix. What message would be relevant? (I will do the update)

This is just to basically verify if enhanced route refresh capability. I guess: `bgp: Check if Enhanced route-refresh capability is printed properly`.

**Comment by fxlb** [comment]
> I guess: bgp: Check if Enhanced route-refresh capability is printed properly.

Good. Done.

**Comment by fxlb** [comment]
Thank you!

**Comment by ton31337** [comment]
@fxlb thanks.

**Pull request merged**
